types/react-test-renderer/react-test-renderer-tests.ts

```diff
@@ -95,3 +95,23 @@ void (async () => {
     // @ts-expect-error
     await act(async () => null);
 })();
+
+[undefined, null, true, false, 0, 1, "", "test"].forEach((children) => {
+    const renderer = create(React.createElement("div", { children }), {
+        createNodeMock: (el: React.ReactElement) => {
+            return {};
+        }
+    });
+
+    const json = renderer.toJSON();
+
+    if (!json || typeof json !== "object" || Array.isArray(json) || json.children !== children) {
+        throw(new Error(`Type of children ${children} doesn't match!`));
+    }
+
+    const tree = renderer.toTree();
+
+    if (!tree || typeof tree !== "object" || Array.isArray(tree) || tree.children !== children) {
+        throw(new Error(`Type of children ${children} doesn't match!`));
+    }
+});
```

Comment:
Could you rewrite this with less JS? It's not clear what is testing types here and what is just for JS.

Reply:
@eps1lon this test asserts a list of primitive values passed as `children`.
It would throw if the `children` of the rendered tree isn't strictly equal to the passed primitve.

Reply:
These tests don't _execute_; they just run the type checker and `$ExpectType` rules.

Reply:
Indeed, none of them assert something meaningful 🤷‍♂️